ddtrace/commands/ddtrace_run.py

```diff
@@ -105,6 +104,9 @@ def main():
 
     log.debug("program executable: %s", executable)
 
+    if "uwsgi" in executable:
+        log.warning("ddtrace-run is not supported with uwsgi. Use manual configuration.")
```

Comment:
hm why is that? I'd think that with the support coded added it'd work, no?

Reply:
Do you know why `import uwsgi` would fail? What's the error/traceback?

When I tested my uwsgi-profiler-patch I didn't encounter this issue IIRC.

Reply:
Here's a repro https://gist.github.com/majorgreys/c57f7b931ecb646d46f22388e4338f52 for the `ModuleNotFoundError`.

```
$ PYTHONPATH=. uwsgi --need-app --die-on-term --http-socket :8888 --wsgi-file uwsgi-app.py --enable-threads --master --processes 2
*** Starting uWSGI 2.0.19.1 (64bit) on [Fri Mar  5 09:18:24 2021] ***
compiled with version: Apple LLVM 12.0.0 (clang-1200.0.32.21) on 04 November 2020 23:40:34
os: Darwin-19.6.0 Darwin Kernel Version 19.6.0: Tue Jan 12 22:13:05 PST 2021; root:xnu-6153.141.16~1/RELEASE_X86_64
nodename: COMP-C02C70L0LVDP
machine: x86_64
clock source: unix
pcre jit disabled
detected number of CPU cores: 8
current working directory: /Users/tahir.butt/dev/datadog/ddtrace-run-uwsgi
detected binary path: /Users/tahir.butt/.virtualenvs/tempenv-43ae25963b2f/bin/uwsgi
your processes number limit is 2784
your memory page size is 4096 bytes
detected max file descriptor number: 256
lock engine: OSX spinlocks
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address :8888 fd 3
Python version: 3.8.6 (default, Oct  5 2020, 12:52:53)  [Clang 12.0.0 (clang-1200.0.32.2)]
Error in sitecustomize; set PYTHONVERBOSE for traceback:
ModuleNotFoundError: No module named 'uwsgi'
Python main interpreter initialized at 0x7fdd31005920
```

Reply:
Thank you!

Damn.

It seems the `uwsgi` module is available once `uwsgi_python_preinit_apps` is called which is after `uwsgi_python_init` and that's likely where `sitecustomize` is imported, so indeed it's too soon.

😭 

So yeah I think that's the right thing to do.